### DISCUSSION TITLE
Micronaut Data equivalent of GORM's DetachedCriteria

### DISCUSSION THREAD

**Discussion by jamesdh** [micronaut-projects/micronaut-core#4985] [Ideas] [answered]
After digging around in the [grails/gorm-graphql ](https://github.com/grails/gorm-graphql)codebase a fair bit, I've come to really appreciate the composability that GORM's DetachedCriteria give you. I'm still trying to wrap my head around parts of that project but how @graemerocher and @jameskleeh used DetachedCriteria for query optimization across deep graph queries is brilliant, and it's easily the best use case I've found for them so far. 
 
Due to Micronaut Data's AoT compilation I have no idea how something similar would be possible, but I'm curious if you've put any thought into anything of the sort?

**Comment by jameskleeh** [comment, marked as answer]
That currently doesn't fit the designed use case for Micronaut data, which is predefined queries. Support for something like that may be added in the future, but for now we recommend tools like jooq.
